src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/readers/JsonReader.java

```diff
@@ -0,0 +1,209 @@
+        if (SECURITY_PRINCIPLES.equals(n)) {
+            this.createPrincipals(o, contentCreator);
+        } else if (SECURITY_ACL.equals(n)) {
+            this.createAclOverride(o, contentCreator);
+        } else {
+            return false;
+        }
+        return true;
+        
+    }
+
+    private void createAclOverride(Object obj, ContentCreator contentCreator) throws RepositoryException {
+        if (obj instanceof JsonObject) {
+            // single ace
+            createAceOverride((JsonObject) obj, contentCreator);
+        } else if (obj instanceof JsonArray) {
+            // array of aces
+            JsonArray jsonArray = (JsonArray) obj;
+            for (int i = 0; i < jsonArray.size(); i++) {
+                Object object = jsonArray.get(i);
+                if (object instanceof JsonObject) {
+                    createAceOverride((JsonObject) object, contentCreator);
+                } else {
+                    throw new JsonException("Unexpected data type in acl array: " + object.getClass().getName());
+                }
+            }
+        }
+    }
+    
+    
+    
+    /**
+     * Create or update an access control entry
+     */
+    private void createAceOverride(JsonObject ace, ContentCreator contentCreator) throws RepositoryException {
+
+        String principalID = ace.getString("principal");
+
+        String [] grantedPrivileges = null;
+        JsonArray granted = (JsonArray) ace.get("granted");
+        if (granted != null) {
+            grantedPrivileges = new String[granted.size()];
+            for (int a=0; a < grantedPrivileges.length; a++) {
+                grantedPrivileges[a] = granted.getString(a);
+            }
+        }
+
+        String [] deniedPrivileges = null;
+        JsonArray denied = (JsonArray) ace.get("denied");
+        if (denied != null) {
+            deniedPrivileges = new String[denied.size()];
+            for (int a=0; a < deniedPrivileges.length; a++) {
+                deniedPrivileges[a] = denied.getString(a);
+            }
+        }
+
+        String order = ace.getString("order", null);
+
+        Map<String, Value> restrictionsMap = new HashMap<>();
+        Map<String, Value[]> mvRestrictionsMap = new HashMap<>();
+        Set<String> removedRestrictionNames = new HashSet<>();
+        
+        if(ace.containsKey("restrictions")){
+            JsonObject restrictions = (JsonObject) ace.get("restrictions");
+
+            Set<String> keySet = restrictions.keySet();
+            for (String rname : keySet) {
+                if (rname.endsWith("@Delete")) {
+                    //add the key to the 'remove' set.  the value doesn't matter and is ignored.
+                    String rname2 = rname.substring(9, rname.length() - 7);
+                    removedRestrictionNames.add(rname2);
+                } else {
+                    RestrictionDefinition rd = RestrictionProvider.get(rname);
```

Comment:
i would just parse the strings and not making any attempt to validate the restrictions. this is not the job of the converter and i cannot know which restrictions are supported by the system (e.g. AEM comes with additional custom restrictionprovider implementations.
-> drop usage of RestrictionDefinition altogether